core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -263,29 +264,30 @@ sealed abstract class Doc extends Product with Serializable {
    * the widest possible variant. This is the same
    * as render(Int.MaxValue) except it is more efficient.
    */
-  def renderWideStream: Stream[String] = {
+  private def renderWideOptionStream: Stream[Option[String]] = {
```

Comment:
I'd rather not use `Stream[Option[String]]` here....

What if we could implement:
```scala
def fails(d: Doc): Boolean  = ...
```
directly as a recursion and not do it by constructing a Stream of options and looking for a None.

Reply:
I dropped `Fail`, so this method is now unchanged, and is stack safe.